File: App.Models/CaseFile.cs
using App.Core.Models;

namespace App.Models
{
    public class CaseFile : IAuditableEntity
    {
        public int Id { get; set; }
        public int ComplaintId { get; set; }
        public int PreparedBy { get; set; }
        public DateTime PreparedAt { get; set; }
        public string Summary { get; set; } = string.Empty;
        public string? Recommendations { get; set; }
        public CaseFileStatus Status { get; set; } = CaseFileStatus.Draft;
        public DateTime CreatedOn { get; set; }
        public DateTime ModifiedOn { get; set; }

        // Navigation properties
        public virtual Complaint Complaint { get; set; } = null!;
        public virtual User Preparer { get; set; } = null!;
        public virtual ICollection<CaseFileDocument> Documents { get; set; } = new List<CaseFileDocument>();
    }
}

File: App.client/projects/proctor-app/scripts/models/case-file.ts
import { User, CaseFileStatus } from './user';
import { Complaint } from './complaint';
import { CaseFileDocument } from './case-file-document';

export class CaseFile {
  id: number;
  complaintId: number;
  preparedBy: number;
  preparedAt: string;
  summary: string;
  recommendations?: string | null;
  status: string;
  createdOn: string;
  modifiedOn: string;
  complaint?: Complaint | null;
  preparer?: User | null;
  documents: CaseFileDocument[];

  constructor() {
    this.id = 0;
    this.complaintId = this.preparedBy = 0;
    this.preparedAt = new Date().toISOString();
    this.summary = '';
    this.recommendations = null;
    this.status = CaseFileStatus.Draft;
    this.createdOn = this.modifiedOn = '';
    this.documents = [];
  }
}

